authz/rbac_translator.go

```diff
@@ -229,36 +222,35 @@ func parseHeaders(headers []header) ([]*v3rbacpb.Permission, error) {
 		if header.Key == "" {
 			return nil, fmt.Errorf(`"headers" %d: "key" is not present`, i)
 		}
-		if unsupportedHeader(header.Key) {
+		if unsupportedHeader(strings.ToLower(header.Key)) {
```

Comment:
Is this the right place to lowercase it?  I think we may want to modify the key in the RBAC config itself, since it will be lowercase in the gRPC metadata.

Reply:
Done. I don't think it is required since in metadata get, the key is converted to lowercase before searching in md.
https://github.com/grpc/grpc-go/blob/master/metadata/metadata.go#L99

Reply:
Looks like we're not using this getter in the RBAC engine, though.  And saving the conversion every time you do a lookup is desirable.